packages/theme-check-common/src/checks/space-around-class_list/index.ts

```diff
@@ -0,0 +1,124 @@
+import { LiquidHtmlNode, NodeTypes } from '@shopify/liquid-html-parser';
+import { LiquidCheckDefinition, Severity, SourceCodeType } from '../../types';
+
+export const SpaceAroundClassList: LiquidCheckDefinition = {
+  meta: {
+    code: 'SpaceAroundClassList',
+    aliases: ['SpaceAroundClassList'],
+    name: 'Space Around Class List',
+    docs: {
+      description: 'Warns you when there is no space before or after using the class_list filter',
+      recommended: true,
+      url: 'https://shopify.dev/docs/themes/tools/theme-check/checks/space-around-class_list',
+    },
+    type: SourceCodeType.LiquidHtml,
+    severity: Severity.ERROR,
+    schema: {},
+    targets: [],
+  },
+
+  create(context) {
+    return {
+      async LiquidFilter(node, ancestors) {
+        if (node.name !== 'class_list') {
+          return;
+        }
+
+        const classAttribute = ancestors.find(
+          (ancestor) =>
+            ancestor.type === NodeTypes.AttrDoubleQuoted ||
+            ancestor.type === NodeTypes.AttrSingleQuoted,
+        );
+
+        if (!classAttribute) {
+          return;
+        }
+
+        const classAttributeContent =
+          classAttribute.source.slice(classAttribute.position.start, classAttribute.position.end) ||
+          '';
+
+        // check for missing space after class_list
+        const afterRegex = /([a-zA-Z0-9._-]+)\s*\|\s*class_list\s*}}([a-zA-Z0-9._-]+)/gm;
```

Comment:
I believe we have valid cases, such as the following one where we want to raise an offense:
```liquid
<div class="{{ block.settings.spacing | class_list }}{{ shop.name }}other-class">content</div>
```

It's a bit changeling to cover that case relying on the regex tho. Instead, I believe we may navigate in the AST and can check if the sibling node starts with a space:

![Screenshot 2024-07-22 at 08 48 34](https://github.com/user-attachments/assets/72d03b57-5dff-4975-9d9f-79df33d703a0)

What do you think?

Reply:
ah, another interesting case. I'm glad y'all know your liquid! Would it be correct then to say the filter's next sibling (if any) must be a TextNode that starts with whitespace?